cli/lsp/performance.rs

```diff
@@ -221,38 +266,32 @@ impl Performance {
     self.mark_inner(name, Some(args))
   }
 
+  /// Creates a performance mark which will be measured against on drop. Use
+  /// like this:
+  /// ```rust
+  /// let _mark = self.performance.measure_scope("foo");
+  /// ```
+  /// Don't use like this:
+  /// ```rust
+  /// // ❌
+  /// let _ = self.performance.measure_scope("foo");
```

Comment:
I wish there were a way in rust for a struct to say that it should never be used like this so that it could be surfaced in a lint rule.